modmesh/apputil.py

```diff
@@ -68,8 +70,32 @@ def __init__(self, name):
         # Each run of the application appends a new environment.
         environ[name] = self
 
+        self._stdoutFd = -1
+        self._stderrFd = -1
+
     def run_code(self, code):
-        exec(code, self.globals, self.locals)
+        import msvcrt
+
+        oldStdoutFd = sys.stdout
+        oldStderrFd = sys.stderr
+        try:
+            if self._stdoutFd <= 0:
+                raise Exception("wrong stdout fd:", self._stdoutFd)
+            if self._stderrFd <= 0:
+                raise Exception("wrong stderr fd:", self._stderrFd)
+
+            sys.stdout = msvcrt.get_osfhandle(self._stdoutFd)
```

Comment:
failed here, although `self._stdoutFd` could be `3` from C++ `fopen`, and `msvcrt.get_osfhandle` translates it to Python's handle, but it seems python cannot take this FD, probably cannot share the same FD for different processes.
```
OSError: [Errno 9] Bad file descriptortraceback:  File "C:\Users\tiger\modmesh\modmesh\system.py", line 105, in enter_main
    ret = _run_viewer(argv)
  File "C:\Users\tiger\modmesh\modmesh\system.py", line 80, in _run_viewer
    return view.launch()
  File "C:\Users\tiger\modmesh\modmesh\view.py", line 95, in launch
    return app.exec()
  File "C:\Users\tiger\modmesh\modmesh\system.py", line 117, in exec_code
    apputil.run_code(code, redirectStdOutFd, redirectStdErrFd)
  File "C:\Users\tiger\modmesh\modmesh\apputil.py", line 129, in run_code
    aenv.run_code(code)
  File "C:\Users\tiger\modmesh\modmesh\apputil.py", line 99, in run_code
    traceback.print_stack()
```

Reply:
Is there a reason that you have been trying to deal with the fd in Python, but not directly in C/C++?  Wouldn't it be easier to simply handle it in CPython API?

Reply:
I haven't tried if I can get a python file object in C++, I could try.